python_modules/dagster/dagster/_core/storage/asset_value_loader.py

```diff
@@ -42,8 +42,12 @@ def __init__(
         self._source_assets_by_key = source_assets_by_key
         self._resource_instance_cache: Dict[str, object] = {}
         self._exit_stack: ExitStack = ExitStack().__enter__()
-        if not instance and is_dagster_home_set():
-            self._instance = self._exit_stack.enter_context(DagsterInstance.get())
+        if not instance:
+            if is_dagster_home_set():
+                instance = DagsterInstance.get()
+            else:
+                instance = DagsterInstance.ephemeral()
```

Comment:
I think in the ephemeral instance case this could be hard to diagnose...

I.e. if you don't have the dagster_home envvar set, but a dynamic partition is existent in your instance, it could be frustrating to keep seeing a "nonexistent partition" error

would it make sense to just raise an error when the envvar isn't set? without access to the correct instance you can't load any meaningful contents, right?